ocaml/xapi/message_forwarding.ml

```diff
@@ -1346,26 +1346,36 @@ functor
         else
           local_fn ~__context
 
-      (* Clear scheduled_to_be_resident_on for a VM and all its vGPUs. *)
-      let clear_scheduled_to_be_resident_on ~__context ~vm =
+      let clear_reservations ~__context ~vm =
+        debug "%s VM=%s" __FUNCTION__ (Ref.string_of vm) ;
+        (* host *)
         Db.VM.set_scheduled_to_be_resident_on ~__context ~self:vm
           ~value:Ref.null ;
-        List.iter
-          (fun vgpu ->
-            Db.VGPU.set_scheduled_to_be_resident_on ~__context ~self:vgpu
-              ~value:Ref.null
-          )
-          (Db.VM.get_VGPUs ~__context ~self:vm)
-
-      let clear_reserved_netsriov_vfs_on ~__context ~vm =
+        (* vgpu *)
+        Db.VM.get_VGPUs ~__context ~self:vm
+        |> List.iter (fun vgpu ->
+               Db.VGPU.set_scheduled_to_be_resident_on ~__context ~self:vgpu
+                 ~value:Ref.null
+           ) ;
+        (* vifs *)
         Db.VM.get_VIFs ~__context ~self:vm
         |> List.iter (fun vif ->
-               let vf = Db.VIF.get_reserved_pci ~__context ~self:vif in
-               Db.VIF.set_reserved_pci ~__context ~self:vif ~value:Ref.null ;
-               if Db.is_valid_ref __context vf then
-                 Db.PCI.set_scheduled_to_be_attached_to ~__context ~self:vf
+               Db.VIF.set_reserved_pci ~__context ~self:vif ~value:Ref.null
+           ) ;
```

Comment:
Looking at how this VIF field is used elsewhere in the code makes me believe that this field is not in the same camp as the `scheduled_to_be` fields. The latter are reset when the VM has successfully started (or resumed, or migrated) and the reservation is moved to the related fields without `scheduled_to` in the name. But `VIF.reserved_pci` seems to be kept while the VM is running. This make be the reason why there was a separate clear function, which was not called in the two `finally` clauses below. `PCI.scheduled_to_be_attached_to` does still need to be cleared even for (SR-IOV) VIFs.

So I think we need to still split out these four lines into a function that is called only in the exception handler of `allocate_vm_to_host`.

Reply:
@robhoes I've addressed this in a fixup that I would squash before merging. BST/BVT were green already.